### DISCUSSION TITLE
version 4.5.0

### DISCUSSION THREAD

**Discussion by hvianna** [hvianna/audioMotion-analyzer#75] [Announcements]
### Added:

+ [`fadePeaks`](README.md#fadepeaks-boolean) - peaks fade out instead of falling down - thanks **@TF3RDL** for this [idea](https://codepen.io/TF3RDL/pen/MWLzPoO);
+ [`gravity`](README.md#gravity-number) - customize peak drop acceleration;
+ [`peakFadeTime`](README.md#peakfadetime-number) - customize peak fade-out time;
+ [`peakHoldTime`](README.md#peakholdtime-number) - customize peak hold time;
+ UMD module export - allows using the module via _require()_ and provides a global variable for easier usage in the browser - thanks **@shahkashani** for the suggestion ([#71](https://github.com/hvianna/audioMotion-analyzer/issues/71) and [#72](https://github.com/hvianna/audioMotion-analyzer/issues/72)); **see the [Usage](README.md#usage) section for more details.**

The new gravity equation restores the original peak drop behavior and decay curve (using the default value) of versions 1.0.0 through 3.4.0.
Check out the [peak drop analysis tool](/tools/peak-drop.html) for more information on the evolution of the equation and to compare different values of `gravity`.

As usual, you can interact with all the new properties in the [fluid demo](/demo/fluid.html).

<hr /><em>This discussion was created from the release <a href='https://github.com/hvianna/audioMotion-analyzer/releases/tag/4.5.0'>version 4.5.0</a>.</em>

**Comment by TF3RDL** [comment]
thanks for bringing the idea of fade out peaks from [analog-style spectrum analyzer/sliding DFT visualization project](https://codepen.io/TF3RDL/pen/MWLzPoO), which I've added it into my aforementioned CodePen project because I saw an audio visualizer thing (basically a peakmeter) in Geometry Dash 2.2

BTW, I think ditching the AnalyserNode's built-in FFT entirely (in favor of custom FFT and other audio analysis functions that don't actually rely on WebAudio at all) per #30 would makes audioMotion-analyzer more feasible for non-realtime analysis and even analysis (both realtime and non-realtime) of non-audio signals in the absence of Web Audio API since the AnalyserNode is designed for realtime analysis and it is a part of Web Audio API, and in-addition, would use the full potential of AnalyserNode.getFloatTimeDomainData and AudioWorklet for realtime audio analysis

**Reply by hvianna** [reply]
Yes, I think it would be great to have these alternate analysis functions in addition to what WebAudio provides.

The first thing I need to work now is on separating the audio data processing from the graphics rendering in the `_draw()` function, so it will be easier to integrate different analysis methods.

I'll be doing this in a new `v5` branch, for the next major version. Would you be willing to help with PRs to bring in your implementation of the custom transforms?

**Reply by TF3RDL** [reply]
True as separating the audio data processing from drawing function can help with integrating custom analysis functions and maybe even makes it easier to add infinite average spectrum in-addition to main bar peaks (much like foo_enhanced_spectrum_analyzer) as in [this updated CodePen project about "analog-style" spectrum analysis](https://codepen.io/TF3RDL/pen/MWLzPoO) right?

BTW, a linear peak decay function as in this CodePen project linked above (and foobar2000's built-in Spectrum and foo_enhanced_spectrum_analyzer visualization) would be great addition to audioMotion-analyzer
